// google-ads-api/build/src/protos/autogen/resourceNames.js
"use strict";
/* Autogenerated File! Do Not Edit */
Object.defineProperty(exports, "__esModule", { value: true });
exports.campaignBudget = exports.campaignBidModifier = exports.campaignAudienceView = exports.campaignAssetSet = exports.campaignAsset = exports.campaign = exports.callView = exports.billingSetup = exports.biddingStrategySimulation = exports.biddingStrategy = exports.biddingSeasonalityAdjustment = exports.biddingDataExclusion = exports.batchJob = exports.audience = exports.assetSetTypeView = exports.assetSetAsset = exports.assetSet = exports.assetGroupSignal = exports.assetGroupProductGroupView = exports.assetGroupListingGroupFilter = exports.assetGroupAsset = exports.assetGroup = exports.assetFieldTypeView = exports.asset = exports.ageRangeView = exports.adScheduleView = exports.adParameter = exports.adGroupSimulation = exports.adGroupLabel = exports.adGroupFeed = exports.adGroupExtensionSetting = exports.adGroupCustomizer = exports.adGroupCriterionSimulation = exports.adGroupCriterionLabel = exports.adGroupCriterionCustomizer = exports.adGroupCriterion = exports.adGroupBidModifier = exports.adGroupAudienceView = exports.adGroupAssetSet = exports.adGroupAsset = exports.adGroupAdLabel = exports.adGroupAdAssetView = exports.adGroupAdAssetCombinationView = exports.adGroupAd = exports.adGroup = exports.ad = exports.accountLink = exports.accountBudgetProposal = exports.accountBudget = exports.accessibleBiddingStrategy = void 0;
exports.experiment = exports.expandedLandingPageView = exports.dynamicSearchAdsSearchTermView = exports.domainCategory = exports.distanceView = exports.displayKeywordView = exports.detailedDemographic = exports.detailPlacementView = exports.customizerAttribute = exports.customerUserAccessInvitation = exports.customerUserAccess = exports.customerSkAdNetworkConversionValueSchema = exports.customerSearchTermInsight = exports.customerNegativeCriterion = exports.customerManagerLink = exports.customerLabel = exports.customerFeed = exports.customerExtensionSetting = exports.customerCustomizer = exports.customerConversionGoal = exports.customerClientLink = exports.customerClient = exports.customerAssetSet = exports.customerAsset = exports.customer = exports.customInterest = exports.customConversionGoal = exports.customAudience = exports.currencyConstant = exports.conversionValueRuleSet = exports.conversionValueRule = exports.conversionGoalCampaignConfig = exports.conversionCustomVariable = exports.conversionAction = exports.combinedAudience = exports.clickView = exports.changeStatus = exports.changeEvent = exports.carrierConstant = exports.campaignSimulation = exports.campaignSharedSet = exports.campaignSearchTermInsight = exports.campaignLabel = exports.campaignGroup = exports.campaignFeed = exports.campaignExtensionSetting = exports.campaignDraft = exports.campaignCustomizer = exports.campaignCriterion = exports.campaignConversionGoal = void 0;
exports.searchTermView = exports.remarketingAction = exports.recommendation = exports.qualifyingQuestion = exports.productLink = exports.productGroupView = exports.productBiddingCategoryConstant = exports.perStoreView = exports.paymentsAccount = exports.parentalStatusView = exports.paidOrganicSearchTermView = exports.operatingSystemVersionConstant = exports.offlineUserDataJob = exports.mobileDeviceConstant = exports.mobileAppCategoryConstant = exports.merchantCenterLink = exports.mediaFile = exports.managedPlacementView = exports.locationView = exports.lifeEvent = exports.leadFormSubmissionData = exports.languageConstant = exports.landingPageView = exports.label = exports.keywordView = exports.keywordThemeConstant = exports.keywordPlanCampaignKeyword = exports.keywordPlanCampaign = exports.keywordPlanAdGroupKeyword = exports.keywordPlanAdGroup = exports.keywordPlan = exports.invoice = exports.incomeRangeView = exports.hotelReconciliation = exports.hotelPerformanceView = exports.hotelGroupView = exports.groupPlacementView = exports.googleAdsField = exports.geographicView = exports.geoTargetConstant = exports.genderView = exports.feedPlaceholderView = exports.feedMapping = exports.feedItemTarget = exports.feedItemSetLink = exports.feedItemSet = exports.feedItem = exports.feed = exports.extensionFeedItem = exports.experimentArm = void 0;
exports.webpageView = exports.video = exports.userLocationView = exports.userList = exports.userInterest = exports.travelActivityPerformanceView = exports.travelActivityGroupView = exports.topicView = exports.topicConstant = exports.thirdPartyAppAnalyticsLink = exports.smartCampaignSetting = exports.smartCampaignSearchTermView = exports.shoppingPerformanceView = exports.sharedSet = exports.sharedCriterion = void 0;
/**
 * @param {string | number} customerId
 * @param {string | number} biddingStrategyId
 * @returns `AccessibleBiddingStrategyResourceName`
 * @example const accessibleBiddingStrategy: ResourceNames.AccessibleBiddingStrategyResourceName = ResourceNames.accessibleBiddingStrategy(10987417, 21974834)
 */
function accessibleBiddingStrategy(customerId, biddingStrategyId) {
    return `customers/${customerId}/accessibleBiddingStrategies/${biddingStrategyId}`;
}
exports.accessibleBiddingStrategy = accessibleBiddingStrategy;
/**
 * @param {string | number} customerId
 * @param {string | number} accountBudgetId
 * @returns `AccountBudgetResourceName`
 * @example const accountBudget: ResourceNames.AccountBudgetResourceName = ResourceNames.accountBudget(10987417, 21974834)
 */
function accountBudget(customerId, accountBudgetId) {
    return `customers/${customerId}/accountBudgets/${accountBudgetId}`;
}
exports.accountBudget = accountBudget;
/**
 * @param {string | number} customerId
 * @param {string | number} accountBudgetProposalId
 * @returns `AccountBudgetProposalResourceName`
 * @example const accountBudgetProposal: ResourceNames.AccountBudgetProposalResourceName = ResourceNames.accountBudgetProposal(10987417, 21974834)
 */
function accountBudgetProposal(customerId, accountBudgetProposalId) {
    return `customers/${customerId}/accountBudgetProposals/${accountBudgetProposalId}`;
}
exports.accountBudgetProposal = accountBudgetProposal;
/**
 * @param {string | number} customerId
 * @param {string | number} accountLinkId
 * @returns `AccountLinkResourceName`
 * @example const accountLink: ResourceNames.AccountLinkResourceName = ResourceNames.accountLink(10987417, 21974834)
 */
function accountLink(customerId, accountLinkId) {
    return `customers/${customerId}/accountLinks/${accountLinkId}`;
}
exports.accountLink = accountLink;
/**
 * @param {string | number} customerId
 * @param {string | number} adId
 * @returns `AdResourceName`
 * @example const ad: ResourceNames.AdResourceName = ResourceNames.ad(10987417, 21974834)
 */
function ad(customerId, adId) {
    return `customers/${customerId}/ads/${adId}`;
}
exports.ad = ad;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @returns `AdGroupResourceName`
 * @example const adGroup: ResourceNames.AdGroupResourceName = ResourceNames.adGroup(10987417, 21974834)
 */
function adGroup(customerId, adGroupId) {
    return `customers/${customerId}/adGroups/${adGroupId}`;
}
exports.adGroup = adGroup;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} adId
 * @returns `AdGroupAdResourceName`
 * @example const adGroupAd: ResourceNames.AdGroupAdResourceName = ResourceNames.adGroupAd(10987417, 21974834, 43949668)
 */
function adGroupAd(customerId, adGroupId, adId) {
    return `customers/${customerId}/adGroupAds/${adGroupId}~${adId}`;
}
exports.adGroupAd = adGroupAd;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} adId
 * @param {string | number} assetCombinationIdLow
 * @param {string | number} assetCombinationIdHigh
 * @returns `AdGroupAdAssetCombinationViewResourceName`
 * @example const adGroupAdAssetCombinationView: ResourceNames.AdGroupAdAssetCombinationViewResourceName = ResourceNames.adGroupAdAssetCombinationView(10987417, 21974834, 43949668, 87899336, 175798672)
 */
function adGroupAdAssetCombinationView(customerId, adGroupId, adId, assetCombinationIdLow, assetCombinationIdHigh) {
    return `customers/${customerId}/adGroupAdAssetCombinationViews/${adGroupId}~${adId}~${assetCombinationIdLow}~${assetCombinationIdHigh}`;
}
exports.adGroupAdAssetCombinationView = adGroupAdAssetCombinationView;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} adId
 * @param {string | number} assetId
 * @param {string | number} fieldType
 * @returns `AdGroupAdAssetViewResourceName`
 * @example const adGroupAdAssetView: ResourceNames.AdGroupAdAssetViewResourceName = ResourceNames.adGroupAdAssetView(10987417, 21974834, 43949668, 87899336, 175798672)
 */
function adGroupAdAssetView(customerId, adGroupId, adId, assetId, fieldType) {
    return `customers/${customerId}/adGroupAdAssetViews/${adGroupId}~${adId}~${assetId}~${fieldType}`;
}
exports.adGroupAdAssetView = adGroupAdAssetView;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} adId
 * @param {string | number} labelId
 * @returns `AdGroupAdLabelResourceName`
 * @example const adGroupAdLabel: ResourceNames.AdGroupAdLabelResourceName = ResourceNames.adGroupAdLabel(10987417, 21974834, 43949668, 87899336)
 */
function adGroupAdLabel(customerId, adGroupId, adId, labelId) {
    return `customers/${customerId}/adGroupAdLabels/${adGroupId}~${adId}~${labelId}`;
}
exports.adGroupAdLabel = adGroupAdLabel;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} assetId
 * @param {string | number} fieldType
 * @returns `AdGroupAssetResourceName`
 * @example const adGroupAsset: ResourceNames.AdGroupAssetResourceName = ResourceNames.adGroupAsset(10987417, 21974834, 43949668, 87899336)
 */
function adGroupAsset(customerId, adGroupId, assetId, fieldType) {
    return `customers/${customerId}/adGroupAssets/${adGroupId}~${assetId}~${fieldType}`;
}
exports.adGroupAsset = adGroupAsset;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} assetSetId
 * @returns `AdGroupAssetSetResourceName`
 * @example const adGroupAssetSet: ResourceNames.AdGroupAssetSetResourceName = ResourceNames.adGroupAssetSet(10987417, 21974834, 43949668)
 */
function adGroupAssetSet(customerId, adGroupId, assetSetId) {
    return `customers/${customerId}/adGroupAssetSets/${adGroupId}~${assetSetId}`;
}
exports.adGroupAssetSet = adGroupAssetSet;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} criterionId
 * @returns `AdGroupAudienceViewResourceName`
 * @example const adGroupAudienceView: ResourceNames.AdGroupAudienceViewResourceName = ResourceNames.adGroupAudienceView(10987417, 21974834, 43949668)
 */
function adGroupAudienceView(customerId, adGroupId, criterionId) {
    return `customers/${customerId}/adGroupAudienceViews/${adGroupId}~${criterionId}`;
}
exports.adGroupAudienceView = adGroupAudienceView;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} criterionId
 * @returns `AdGroupBidModifierResourceName`
 * @example const adGroupBidModifier: ResourceNames.AdGroupBidModifierResourceName = ResourceNames.adGroupBidModifier(10987417, 21974834, 43949668)
 */
function adGroupBidModifier(customerId, adGroupId, criterionId) {
    return `customers/${customerId}/adGroupBidModifiers/${adGroupId}~${criterionId}`;
}
exports.adGroupBidModifier = adGroupBidModifier;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} criterionId
 * @returns `AdGroupCriterionResourceName`
 * @example const adGroupCriterion: ResourceNames.AdGroupCriterionResourceName = ResourceNames.adGroupCriterion(10987417, 21974834, 43949668)
 */
function adGroupCriterion(customerId, adGroupId, criterionId) {
    return `customers/${customerId}/adGroupCriteria/${adGroupId}~${criterionId}`;
}
exports.adGroupCriterion = adGroupCriterion;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} criterionId
 * @param {string | number} customizerAttributeId
 * @returns `AdGroupCriterionCustomizerResourceName`
 * @example const adGroupCriterionCustomizer: ResourceNames.AdGroupCriterionCustomizerResourceName = ResourceNames.adGroupCriterionCustomizer(10987417, 21974834, 43949668, 87899336)
 */
function adGroupCriterionCustomizer(customerId, adGroupId, criterionId, customizerAttributeId) {
    return `customers/${customerId}/adGroupCriterionCustomizers/${adGroupId}~${criterionId}~${customizerAttributeId}`;
}
exports.adGroupCriterionCustomizer = adGroupCriterionCustomizer;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} criterionId
 * @param {string | number} labelId
 * @returns `AdGroupCriterionLabelResourceName`
 * @example const adGroupCriterionLabel: ResourceNames.AdGroupCriterionLabelResourceName = ResourceNames.adGroupCriterionLabel(10987417, 21974834, 43949668, 87899336)
 */
function adGroupCriterionLabel(customerId, adGroupId, criterionId, labelId) {
    return `customers/${customerId}/adGroupCriterionLabels/${adGroupId}~${criterionId}~${labelId}`;
}
exports.adGroupCriterionLabel = adGroupCriterionLabel;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} criterionId
 * @param {string | number} type
 * @param {string | number} modificationMethod
 * @param {string | number} startDate
 * @param {string | number} endDate
 * @returns `AdGroupCriterionSimulationResourceName`
 * @example const adGroupCriterionSimulation: ResourceNames.AdGroupCriterionSimulationResourceName = ResourceNames.adGroupCriterionSimulation(10987417, 21974834, 43949668, 87899336, 175798672, 351597344, 703194688)
 */
function adGroupCriterionSimulation(customerId, adGroupId, criterionId, type, modificationMethod, startDate, endDate) {
    return `customers/${customerId}/adGroupCriterionSimulations/${adGroupId}~${criterionId}~${type}~${modificationMethod}~${startDate}~${endDate}`;
}
exports.adGroupCriterionSimulation = adGroupCriterionSimulation;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} customizerAttributeId
 * @returns `AdGroupCustomizerResourceName`
 * @example const adGroupCustomizer: ResourceNames.AdGroupCustomizerResourceName = ResourceNames.adGroupCustomizer(10987417, 21974834, 43949668)
 */
function adGroupCustomizer(customerId, adGroupId, customizerAttributeId) {
    return `customers/${customerId}/adGroupCustomizers/${adGroupId}~${customizerAttributeId}`;
}
exports.adGroupCustomizer = adGroupCustomizer;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} extensionType
 * @returns `AdGroupExtensionSettingResourceName`
 * @example const adGroupExtensionSetting: ResourceNames.AdGroupExtensionSettingResourceName = ResourceNames.adGroupExtensionSetting(10987417, 21974834, 43949668)
 */
function adGroupExtensionSetting(customerId, adGroupId, extensionType) {
    return `customers/${customerId}/adGroupExtensionSettings/${adGroupId}~${extensionType}`;
}
exports.adGroupExtensionSetting = adGroupExtensionSetting;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} feedId
 * @returns `AdGroupFeedResourceName`
 * @example const adGroupFeed: ResourceNames.AdGroupFeedResourceName = ResourceNames.adGroupFeed(10987417, 21974834, 43949668)
 */
function adGroupFeed(customerId, adGroupId, feedId) {
    return `customers/${customerId}/adGroupFeeds/${adGroupId}~${feedId}`;
}
exports.adGroupFeed = adGroupFeed;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} labelId
 * @returns `AdGroupLabelResourceName`
 * @example const adGroupLabel: ResourceNames.AdGroupLabelResourceName = ResourceNames.adGroupLabel(10987417, 21974834, 43949668)
 */
function adGroupLabel(customerId, adGroupId, labelId) {
    return `customers/${customerId}/adGroupLabels/${adGroupId}~${labelId}`;
}
exports.adGroupLabel = adGroupLabel;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} type
 * @param {string | number} modificationMethod
 * @param {string | number} startDate
 * @param {string | number} endDate
 * @returns `AdGroupSimulationResourceName`
 * @example const adGroupSimulation: ResourceNames.AdGroupSimulationResourceName = ResourceNames.adGroupSimulation(10987417, 21974834, 43949668, 87899336, 175798672, 351597344)
 */
function adGroupSimulation(customerId, adGroupId, type, modificationMethod, startDate, endDate) {
    return `customers/${customerId}/adGroupSimulations/${adGroupId}~${type}~${modificationMethod}~${startDate}~${endDate}`;
}
exports.adGroupSimulation = adGroupSimulation;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} criterionId
 * @param {string | number} parameterIndex
 * @returns `AdParameterResourceName`
 * @example const adParameter: ResourceNames.AdParameterResourceName = ResourceNames.adParameter(10987417, 21974834, 43949668, 87899336)
 */
function adParameter(customerId, adGroupId, criterionId, parameterIndex) {
    return `customers/${customerId}/adParameters/${adGroupId}~${criterionId}~${parameterIndex}`;
}
exports.adParameter = adParameter;
/**
 * @param {string | number} customerId
 * @param {string | number} campaignId
 * @param {string | number} criterionId
 * @returns `AdScheduleViewResourceName`
 * @example const adScheduleView: ResourceNames.AdScheduleViewResourceName = ResourceNames.adScheduleView(10987417, 21974834, 43949668)
 */
function adScheduleView(customerId, campaignId, criterionId) {
    return `customers/${customerId}/adScheduleViews/${campaignId}~${criterionId}`;
}
exports.adScheduleView = adScheduleView;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} criterionId
 * @returns `AgeRangeViewResourceName`
 * @example const ageRangeView: ResourceNames.AgeRangeViewResourceName = ResourceNames.ageRangeView(10987417, 21974834, 43949668)
 */
function ageRangeView(customerId, adGroupId, criterionId) {
    return `customers/${customerId}/ageRangeViews/${adGroupId}~${criterionId}`;
}
exports.ageRangeView = ageRangeView;
/**
 * @param {string | number} customerId
 * @param {string | number} assetId
 * @returns `AssetResourceName`
 * @example const asset: ResourceNames.AssetResourceName = ResourceNames.asset(10987417, 21974834)
 */
function asset(customerId, assetId) {
    return `customers/${customerId}/assets/${assetId}`;
}
exports.asset = asset;
/**
 * @param {string | number} customerId
 * @param {string | number} fieldType
 * @returns `AssetFieldTypeViewResourceName`
 * @example const assetFieldTypeView: ResourceNames.AssetFieldTypeViewResourceName = ResourceNames.assetFieldTypeView(10987417, 21974834)
 */
function assetFieldTypeView(customerId, fieldType) {
    return `customers/${customerId}/assetFieldTypeViews/${fieldType}`;
}
exports.assetFieldTypeView = assetFieldTypeView;
/**
 * @param {string | number} customerId
 * @param {string | number} assetGroupId
 * @returns `AssetGroupResourceName`
 * @example const assetGroup: ResourceNames.AssetGroupResourceName = ResourceNames.assetGroup(10987417, 21974834)
 */
function assetGroup(customerId, assetGroupId) {
    return `customers/${customerId}/assetGroups/${assetGroupId}`;
}
exports.assetGroup = assetGroup;
/**
 * @param {string | number} customerId
 * @param {string | number} assetGroupId
 * @param {string | number} assetId
 * @param {string | number} fieldType
 * @returns `AssetGroupAssetResourceName`
 * @example const assetGroupAsset: ResourceNames.AssetGroupAssetResourceName = ResourceNames.assetGroupAsset(10987417, 21974834, 43949668, 87899336)
 */
function assetGroupAsset(customerId, assetGroupId, assetId, fieldType) {
    return `customers/${customerId}/assetGroupAssets/${assetGroupId}~${assetId}~${fieldType}`;
}
exports.assetGroupAsset = assetGroupAsset;
/**
 * @param {string | number} customerId
 * @param {string | number} assetGroupId
 * @param {string | number} listingGroupFilterId
 * @returns `AssetGroupListingGroupFilterResourceName`
 * @example const assetGroupListingGroupFilter: ResourceNames.AssetGroupListingGroupFilterResourceName = ResourceNames.assetGroupListingGroupFilter(10987417, 21974834, 43949668)
 */
function assetGroupListingGroupFilter(customerId, assetGroupId, listingGroupFilterId) {
    return `customers/${customerId}/assetGroupListingGroupFilters/${assetGroupId}~${listingGroupFilterId}`;
}
exports.assetGroupListingGroupFilter = assetGroupListingGroupFilter;
/**
 * @param {string | number} customerId
 * @param {string | number} assetGroupId
 * @param {string | number} listingGroupFilterId
 * @returns `AssetGroupProductGroupViewResourceName`
 * @example const assetGroupProductGroupView: ResourceNames.AssetGroupProductGroupViewResourceName = ResourceNames.assetGroupProductGroupView(10987417, 21974834, 43949668)
 */
function assetGroupProductGroupView(customerId, assetGroupId, listingGroupFilterId) {
    return `customers/${customerId}/assetGroupProductGroupViews/${assetGroupId}~${listingGroupFilterId}`;
}
exports.assetGroupProductGroupView = assetGroupProductGroupView;
/**
 * @param {string | number} customerId
 * @param {string | number} assetGroupId
 * @param {string | number} criterionId
 * @returns `AssetGroupSignalResourceName`
 * @example const assetGroupSignal: ResourceNames.AssetGroupSignalResourceName = ResourceNames.assetGroupSignal(10987417, 21974834, 43949668)
 */
function assetGroupSignal(customerId, assetGroupId, criterionId) {
    return `customers/${customerId}/assetGroupSignals/${assetGroupId}~${criterionId}`;
}
exports.assetGroupSignal = assetGroupSignal;
/**
 * @param {string | number} customerId
 * @param {string | number} assetSetId
 * @returns `AssetSetResourceName`
 * @example const assetSet: ResourceNames.AssetSetResourceName = ResourceNames.assetSet(10987417, 21974834)
 */
function assetSet(customerId, assetSetId) {
    return `customers/${customerId}/assetSets/${assetSetId}`;
}
exports.assetSet = assetSet;
/**
 * @param {string | number} customerId
 * @param {string | number} assetSetId
 * @param {string | number} assetId
 * @returns `AssetSetAssetResourceName`
 * @example const assetSetAsset: ResourceNames.AssetSetAssetResourceName = ResourceNames.assetSetAsset(10987417, 21974834, 43949668)
 */
function assetSetAsset(customerId, assetSetId, assetId) {
    return `customers/${customerId}/assetSetAssets/${assetSetId}~${assetId}`;
}
exports.assetSetAsset = assetSetAsset;
/**
 * @param {string | number} customerId
 * @param {string | number} assetSetType
 * @returns `AssetSetTypeViewResourceName`
 * @example const assetSetTypeView: ResourceNames.AssetSetTypeViewResourceName = ResourceNames.assetSetTypeView(10987417, 21974834)
 */
function assetSetTypeView(customerId, assetSetType) {
    return `customers/${customerId}/assetSetTypeViews/${assetSetType}`;
}
exports.assetSetTypeView = assetSetTypeView;
/**
 * @param {string | number} customerId
 * @param {string | number} audienceId
 * @returns `AudienceResourceName`
 * @example const audience: ResourceNames.AudienceResourceName = ResourceNames.audience(10987417, 21974834)
 */
function audience(customerId, audienceId) {
    return `customers/${customerId}/audiences/${audienceId}`;
}
exports.audience = audience;
/**
 * @param {string | number} customerId
 * @param {string | number} batchJobId
 * @returns `BatchJobResourceName`
 * @example const batchJob: ResourceNames.BatchJobResourceName = ResourceNames.batchJob(10987417, 21974834)
 */
function batchJob(customerId, batchJobId) {
    return `customers/${customerId}/batchJobs/${batchJobId}`;
}
exports.batchJob = batchJob;
/**
 * @param {string | number} customerId
 * @param {string | number} seasonalityEventId
 * @returns `BiddingDataExclusionResourceName`
 * @example const biddingDataExclusion: ResourceNames.BiddingDataExclusionResourceName = ResourceNames.biddingDataExclusion(10987417, 21974834)
 */
function biddingDataExclusion(customerId, seasonalityEventId) {
    return `customers/${customerId}/biddingDataExclusions/${seasonalityEventId}`;
}
exports.biddingDataExclusion = biddingDataExclusion;
/**
 * @param {string | number} customerId
 * @param {string | number} seasonalityEventId
 * @returns `BiddingSeasonalityAdjustmentResourceName`
 * @example const biddingSeasonalityAdjustment: ResourceNames.BiddingSeasonalityAdjustmentResourceName = ResourceNames.biddingSeasonalityAdjustment(10987417, 21974834)
 */
function biddingSeasonalityAdjustment(customerId, seasonalityEventId) {
    return `customers/${customerId}/biddingSeasonalityAdjustments/${seasonalityEventId}`;
}
exports.biddingSeasonalityAdjustment = biddingSeasonalityAdjustment;
/**
 * @param {string | number} customerId
 * @param {string | number} biddingStrategyId
 * @returns `BiddingStrategyResourceName`
 * @example const biddingStrategy: ResourceNames.BiddingStrategyResourceName = ResourceNames.biddingStrategy(10987417, 21974834)
 */
function biddingStrategy(customerId, biddingStrategyId) {
    return `customers/${customerId}/biddingStrategies/${biddingStrategyId}`;
}
exports.biddingStrategy = biddingStrategy;
/**
 * @param {string | number} customerId
 * @param {string | number} biddingStrategyId
 * @param {string | number} type
 * @param {string | number} modificationMethod
 * @param {string | number} startDate
 * @param {string | number} endDate
 * @returns `BiddingStrategySimulationResourceName`
 * @example const biddingStrategySimulation: ResourceNames.BiddingStrategySimulationResourceName = ResourceNames.biddingStrategySimulation(10987417, 21974834, 43949668, 87899336, 175798672, 351597344)
 */
function biddingStrategySimulation(customerId, biddingStrategyId, type, modificationMethod, startDate, endDate) {
    return `customers/${customerId}/biddingStrategySimulations/${biddingStrategyId}~${type}~${modificationMethod}~${startDate}~${endDate}`;
}
exports.biddingStrategySimulation = biddingStrategySimulation;
/**
 * @param {string | number} customerId
 * @param {string | number} billingSetupId
 * @returns `BillingSetupResourceName`
 * @example const billingSetup: ResourceNames.BillingSetupResourceName = ResourceNames.billingSetup(10987417, 21974834)
 */
function billingSetup(customerId, billingSetupId) {
    return `customers/${customerId}/billingSetups/${billingSetupId}`;
}
exports.billingSetup = billingSetup;
/**
 * @param {string | number} customerId
 * @param {string | number} callDetailId
 * @returns `CallViewResourceName`
 * @example const callView: ResourceNames.CallViewResourceName = ResourceNames.callView(10987417, 21974834)
 */
function callView(customerId, callDetailId) {
    return `customers/${customerId}/callViews/${callDetailId}`;
}
exports.callView = callView;
/**
 * @param {string | number} customerId
 * @param {string | number} campaignId
 * @returns `CampaignResourceName`
 * @example const campaign: ResourceNames.CampaignResourceName = ResourceNames.campaign(10987417, 21974834)
 */
function campaign(customerId, campaignId) {
    return `customers/${customerId}/campaigns/${campaignId}`;
}
exports.campaign = campaign;
/**
 * @param {string | number} customerId
 * @param {string | number} campaignId
 * @param {string | number} assetId
 * @param {string | number} fieldType
 * @returns `CampaignAssetResourceName`
 * @example const campaignAsset: ResourceNames.CampaignAssetResourceName = ResourceNames.campaignAsset(10987417, 21974834, 43949668, 87899336)
 */
function campaignAsset(customerId, campaignId, assetId, fieldType) {
    return `customers/${customerId}/campaignAssets/${campaignId}~${assetId}~${fieldType}`;
}
exports.campaignAsset = campaignAsset;
/**
 * @param {string | number} customerId
 * @param {string | number} campaignId
 * @param {string | number} assetSetId
 * @returns `CampaignAssetSetResourceName`
 * @example const campaignAssetSet: ResourceNames.CampaignAssetSetResourceName = ResourceNames.campaignAssetSet(10987417, 21974834, 43949668)
 */
function campaignAssetSet(customerId, campaignId, assetSetId) {
    return `customers/${customerId}/campaignAssetSets/${campaignId}~${assetSetId}`;
}
exports.campaignAssetSet = campaignAssetSet;
/**
 * @param {string | number} customerId
 * @param {string | number} campaignId
 * @param {string | number} criterionId
 * @returns `CampaignAudienceViewResourceName`
 * @example const campaignAudienceView: ResourceNames.CampaignAudienceViewResourceName = ResourceNames.campaignAudienceView(10987417, 21974834, 43949668)
 */
function campaignAudienceView(customerId, campaignId, criterionId) {
    return `customers/${customerId}/campaignAudienceViews/${campaignId}~${criterionId}`;
}
exports.campaignAudienceView = campaignAudienceView;
/**
 * @param {string | number} customerId
 * @param {string | number} campaignId
 * @param {string | number} criterionId
 * @returns `CampaignBidModifierResourceName`
 * @example const campaignBidModifier: ResourceNames.CampaignBidModifierResourceName = ResourceNames.campaignBidModifier(10987417, 21974834, 43949668)
 */
function campaignBidModifier(customerId, campaignId, criterionId) {
    return `customers/${customerId}/campaignBidModifiers/${campaignId}~${criterionId}`;
}
exports.campaignBidModifier = campaignBidModifier;
/**
 * @param {string | number} customerId
 * @param {string | number} campaignBudgetId
 * @returns `CampaignBudgetResourceName`
 * @example const campaignBudget: ResourceNames.CampaignBudgetResourceName = ResourceNames.campaignBudget(10987417, 21974834)
 */
function campaignBudget(customerId, campaignBudgetId) {
    return `customers/${customerId}/campaignBudgets/${campaignBudgetId}`;
}
exports.campaignBudget = campaignBudget;
/**
 * @param {string | number} customerId
 * @param {string | number} campaignId
 * @param {string | number} category
 * @param {string | number} source
 * @returns `CampaignConversionGoalResourceName`
 * @example const campaignConversionGoal: ResourceNames.CampaignConversionGoalResourceName = ResourceNames.campaignConversionGoal(10987417, 21974834, 43949668, 87899336)
 */
function campaignConversionGoal(customerId, campaignId, category, source) {
    return `customers/${customerId}/campaignConversionGoals/${campaignId}~${category}~${source}`;
}
exports.campaignConversionGoal = campaignConversionGoal;
/**
 * @param {string | number} customerId
 * @param {string | number} campaignId
 * @param {string | number} criterionId
 * @returns `CampaignCriterionResourceName`
 * @example const campaignCriterion: ResourceNames.CampaignCriterionResourceName = ResourceNames.campaignCriterion(10987417, 21974834, 43949668)
 */
function campaignCriterion(customerId, campaignId, criterionId) {
    return `customers/${customerId}/campaignCriteria/${campaignId}~${criterionId}`;
}
exports.campaignCriterion = campaignCriterion;
/**
 * @param {string | number} customerId
 * @param {string | number} campaignId
 * @param {string | number} customizerAttributeId
 * @returns `CampaignCustomizerResourceName`
 * @example const campaignCustomizer: ResourceNames.CampaignCustomizerResourceName = ResourceNames.campaignCustomizer(10987417, 21974834, 43949668)
 */
function campaignCustomizer(customerId, campaignId, customizerAttributeId) {
    return `customers/${customerId}/campaignCustomizers/${campaignId}~${customizerAttributeId}`;
}
exports.campaignCustomizer = campaignCustomizer;
/**
 * @param {string | number} customerId
 * @param {string | number} baseCampaignId
 * @param {string | number} draftId
 * @returns `CampaignDraftResourceName`
 * @example const campaignDraft: ResourceNames.CampaignDraftResourceName = ResourceNames.campaignDraft(10987417, 21974834, 43949668)
 */
function campaignDraft(customerId, baseCampaignId, draftId) {
    return `customers/${customerId}/campaignDrafts/${baseCampaignId}~${draftId}`;
}
exports.campaignDraft = campaignDraft;
/**
 * @param {string | number} customerId
 * @param {string | number} campaignId
 * @param {string | number} extensionType
 * @returns `CampaignExtensionSettingResourceName`
 * @example const campaignExtensionSetting: ResourceNames.CampaignExtensionSettingResourceName = ResourceNames.campaignExtensionSetting(10987417, 21974834, 43949668)
 */
function campaignExtensionSetting(customerId, campaignId, extensionType) {
    return `customers/${customerId}/campaignExtensionSettings/${campaignId}~${extensionType}`;
}
exports.campaignExtensionSetting = campaignExtensionSetting;
/**
 * @param {string | number} customerId
 * @param {string | number} campaignId
 * @param {string | number} feedId
 * @returns `CampaignFeedResourceName`
 * @example const campaignFeed: ResourceNames.CampaignFeedResourceName = ResourceNames.campaignFeed(10987417, 21974834, 43949668)
 */
function campaignFeed(customerId, campaignId, feedId) {
    return `customers/${customerId}/campaignFeeds/${campaignId}~${feedId}`;
}
exports.campaignFeed = campaignFeed;
/**
 * @param {string | number} customerId
 * @param {string | number} campaignGroupId
 * @returns `CampaignGroupResourceName`
 * @example const campaignGroup: ResourceNames.CampaignGroupResourceName = ResourceNames.campaignGroup(10987417, 21974834)
 */
function campaignGroup(customerId, campaignGroupId) {
    return `customers/${customerId}/campaignGroups/${campaignGroupId}`;
}
exports.campaignGroup = campaignGroup;
/**
 * @param {string | number} customerId
 * @param {string | number} campaignId
 * @param {string | number} labelId
 * @returns `CampaignLabelResourceName`
 * @example const campaignLabel: ResourceNames.CampaignLabelResourceName = ResourceNames.campaignLabel(10987417, 21974834, 43949668)
 */
function campaignLabel(customerId, campaignId, labelId) {
    return `customers/${customerId}/campaignLabels/${campaignId}~${labelId}`;
}
exports.campaignLabel = campaignLabel;
/**
 * @param {string | number} customerId
 * @param {string | number} campaignId
 * @param {string | number} clusterId
 * @returns `CampaignSearchTermInsightResourceName`
 * @example const campaignSearchTermInsight: ResourceNames.CampaignSearchTermInsightResourceName = ResourceNames.campaignSearchTermInsight(10987417, 21974834, 43949668)
 */
function campaignSearchTermInsight(customerId, campaignId, clusterId) {
    return `customers/${customerId}/campaignSearchTermInsights/${campaignId}~${clusterId}`;
}
exports.campaignSearchTermInsight = campaignSearchTermInsight;
/**
 * @param {string | number} customerId
 * @param {string | number} campaignId
 * @param {string | number} sharedSetId
 * @returns `CampaignSharedSetResourceName`
 * @example const campaignSharedSet: ResourceNames.CampaignSharedSetResourceName = ResourceNames.campaignSharedSet(10987417, 21974834, 43949668)
 */
function campaignSharedSet(customerId, campaignId, sharedSetId) {
    return `customers/${customerId}/campaignSharedSets/${campaignId}~${sharedSetId}`;
}
exports.campaignSharedSet = campaignSharedSet;
/**
 * @param {string | number} customerId
 * @param {string | number} campaignId
 * @param {string | number} type
 * @param {string | number} modificationMethod
 * @param {string | number} startDate
 * @param {string | number} endDate
 * @returns `CampaignSimulationResourceName`
 * @example const campaignSimulation: ResourceNames.CampaignSimulationResourceName = ResourceNames.campaignSimulation(10987417, 21974834, 43949668, 87899336, 175798672, 351597344)
 */
function campaignSimulation(customerId, campaignId, type, modificationMethod, startDate, endDate) {
    return `customers/${customerId}/campaignSimulations/${campaignId}~${type}~${modificationMethod}~${startDate}~${endDate}`;
}
exports.campaignSimulation = campaignSimulation;
/**
 * @param {string | number} criterionId
 * @returns `CarrierConstantResourceName`
 * @example const carrierConstant: ResourceNames.CarrierConstantResourceName = ResourceNames.carrierConstant(10987417)
 */
function carrierConstant(criterionId) {
    return `carrierConstants/${criterionId}`;
}
exports.carrierConstant = carrierConstant;
/**
 * @param {string | number} customerId
 * @param {string | number} timestampMicros
 * @param {string | number} commandIndex
 * @param {string | number} mutateIndex
 * @returns `ChangeEventResourceName`
 * @example const changeEvent: ResourceNames.ChangeEventResourceName = ResourceNames.changeEvent(10987417, 21974834, 43949668, 87899336)
 */
function changeEvent(customerId, timestampMicros, commandIndex, mutateIndex) {
    return `customers/${customerId}/changeEvents/${timestampMicros}~${commandIndex}~${mutateIndex}`;
}
exports.changeEvent = changeEvent;
/**
 * @param {string | number} customerId
 * @param {string | number} changeStatusId
 * @returns `ChangeStatusResourceName`
 * @example const changeStatus: ResourceNames.ChangeStatusResourceName = ResourceNames.changeStatus(10987417, 21974834)
 */
function changeStatus(customerId, changeStatusId) {
    return `customers/${customerId}/changeStatus/${changeStatusId}`;
}
exports.changeStatus = changeStatus;
/**
 * @param {string | number} customerId
 * @param {string | number} date
 * @param {string | number} gclid
 * @returns `ClickViewResourceName`
 * @example const clickView: ResourceNames.ClickViewResourceName = ResourceNames.clickView(10987417, 21974834, 43949668)
 */
function clickView(customerId, date, gclid) {
    return `customers/${customerId}/clickViews/${date}~${gclid}`;
}
exports.clickView = clickView;
/**
 * @param {string | number} customerId
 * @param {string | number} combinedAudienceId
 * @returns `CombinedAudienceResourceName`
 * @example const combinedAudience: ResourceNames.CombinedAudienceResourceName = ResourceNames.combinedAudience(10987417, 21974834)
 */
function combinedAudience(customerId, combinedAudienceId) {
    return `customers/${customerId}/combinedAudiences/${combinedAudienceId}`;
}
exports.combinedAudience = combinedAudience;
/**
 * @param {string | number} customerId
 * @param {string | number} conversionActionId
 * @returns `ConversionActionResourceName`
 * @example const conversionAction: ResourceNames.ConversionActionResourceName = ResourceNames.conversionAction(10987417, 21974834)
 */
function conversionAction(customerId, conversionActionId) {
    return `customers/${customerId}/conversionActions/${conversionActionId}`;
}
exports.conversionAction = conversionAction;
/**
 * @param {string | number} customerId
 * @param {string | number} conversionCustomVariableId
 * @returns `ConversionCustomVariableResourceName`
 * @example const conversionCustomVariable: ResourceNames.ConversionCustomVariableResourceName = ResourceNames.conversionCustomVariable(10987417, 21974834)
 */
function conversionCustomVariable(customerId, conversionCustomVariableId) {
    return `customers/${customerId}/conversionCustomVariables/${conversionCustomVariableId}`;
}
exports.conversionCustomVariable = conversionCustomVariable;
/**
 * @param {string | number} customerId
 * @param {string | number} campaignId
 * @returns `ConversionGoalCampaignConfigResourceName`
 * @example const conversionGoalCampaignConfig: ResourceNames.ConversionGoalCampaignConfigResourceName = ResourceNames.conversionGoalCampaignConfig(10987417, 21974834)
 */
function conversionGoalCampaignConfig(customerId, campaignId) {
    return `customers/${customerId}/conversionGoalCampaignConfigs/${campaignId}`;
}
exports.conversionGoalCampaignConfig = conversionGoalCampaignConfig;
/**
 * @param {string | number} customerId
 * @param {string | number} conversionValueRuleId
 * @returns `ConversionValueRuleResourceName`
 * @example const conversionValueRule: ResourceNames.ConversionValueRuleResourceName = ResourceNames.conversionValueRule(10987417, 21974834)
 */
function conversionValueRule(customerId, conversionValueRuleId) {
    return `customers/${customerId}/conversionValueRules/${conversionValueRuleId}`;
}
exports.conversionValueRule = conversionValueRule;
/**
 * @param {string | number} customerId
 * @param {string | number} conversionValueRuleSetId
 * @returns `ConversionValueRuleSetResourceName`
 * @example const conversionValueRuleSet: ResourceNames.ConversionValueRuleSetResourceName = ResourceNames.conversionValueRuleSet(10987417, 21974834)
 */
function conversionValueRuleSet(customerId, conversionValueRuleSetId) {
    return `customers/${customerId}/conversionValueRuleSets/${conversionValueRuleSetId}`;
}
exports.conversionValueRuleSet = conversionValueRuleSet;
/**
 * @param {string | number} code
 * @returns `CurrencyConstantResourceName`
 * @example const currencyConstant: ResourceNames.CurrencyConstantResourceName = ResourceNames.currencyConstant(10987417)
 */
function currencyConstant(code) {
    return `currencyConstants/${code}`;
}
exports.currencyConstant = currencyConstant;
/**
 * @param {string | number} customerId
 * @param {string | number} customAudienceId
 * @returns `CustomAudienceResourceName`
 * @example const customAudience: ResourceNames.CustomAudienceResourceName = ResourceNames.customAudience(10987417, 21974834)
 */
function customAudience(customerId, customAudienceId) {
    return `customers/${customerId}/customAudiences/${customAudienceId}`;
}
exports.customAudience = customAudience;
/**
 * @param {string | number} customerId
 * @param {string | number} goalId
 * @returns `CustomConversionGoalResourceName`
 * @example const customConversionGoal: ResourceNames.CustomConversionGoalResourceName = ResourceNames.customConversionGoal(10987417, 21974834)
 */
function customConversionGoal(customerId, goalId) {
    return `customers/${customerId}/customConversionGoals/${goalId}`;
}
exports.customConversionGoal = customConversionGoal;
/**
 * @param {string | number} customerId
 * @param {string | number} customInterestId
 * @returns `CustomInterestResourceName`
 * @example const customInterest: ResourceNames.CustomInterestResourceName = ResourceNames.customInterest(10987417, 21974834)
 */
function customInterest(customerId, customInterestId) {
    return `customers/${customerId}/customInterests/${customInterestId}`;
}
exports.customInterest = customInterest;
/**
 * @param {string | number} customerId
 * @returns `CustomerResourceName`
 * @example const customer: ResourceNames.CustomerResourceName = ResourceNames.customer(10987417)
 */
function customer(customerId) {
    return `customers/${customerId}`;
}
exports.customer = customer;
/**
 * @param {string | number} customerId
 * @param {string | number} assetId
 * @param {string | number} fieldType
 * @returns `CustomerAssetResourceName`
 * @example const customerAsset: ResourceNames.CustomerAssetResourceName = ResourceNames.customerAsset(10987417, 21974834, 43949668)
 */
function customerAsset(customerId, assetId, fieldType) {
    return `customers/${customerId}/customerAssets/${assetId}~${fieldType}`;
}
exports.customerAsset = customerAsset;
/**
 * @param {string | number} customerId
 * @param {string | number} assetSetId
 * @returns `CustomerAssetSetResourceName`
 * @example const customerAssetSet: ResourceNames.CustomerAssetSetResourceName = ResourceNames.customerAssetSet(10987417, 21974834)
 */
function customerAssetSet(customerId, assetSetId) {
    return `customers/${customerId}/customerAssetSets/${assetSetId}`;
}
exports.customerAssetSet = customerAssetSet;
/**
 * @param {string | number} customerId
 * @param {string | number} clientCustomerId
 * @returns `CustomerClientResourceName`
 * @example const customerClient: ResourceNames.CustomerClientResourceName = ResourceNames.customerClient(10987417, 21974834)
 */
function customerClient(customerId, clientCustomerId) {
    return `customers/${customerId}/customerClients/${clientCustomerId}`;
}
exports.customerClient = customerClient;
/**
 * @param {string | number} customerId
 * @param {string | number} clientCustomerId
 * @param {string | number} managerLinkId
 * @returns `CustomerClientLinkResourceName`
 * @example const customerClientLink: ResourceNames.CustomerClientLinkResourceName = ResourceNames.customerClientLink(10987417, 21974834, 43949668)
 */
function customerClientLink(customerId, clientCustomerId, managerLinkId) {
    return `customers/${customerId}/customerClientLinks/${clientCustomerId}~${managerLinkId}`;
}
exports.customerClientLink = customerClientLink;
/**
 * @param {string | number} customerId
 * @param {string | number} category
 * @param {string | number} source
 * @returns `CustomerConversionGoalResourceName`
 * @example const customerConversionGoal: ResourceNames.CustomerConversionGoalResourceName = ResourceNames.customerConversionGoal(10987417, 21974834, 43949668)
 */
function customerConversionGoal(customerId, category, source) {
    return `customers/${customerId}/customerConversionGoals/${category}~${source}`;
}
exports.customerConversionGoal = customerConversionGoal;
/**
 * @param {string | number} customerId
 * @param {string | number} customizerAttributeId
 * @returns `CustomerCustomizerResourceName`
 * @example const customerCustomizer: ResourceNames.CustomerCustomizerResourceName = ResourceNames.customerCustomizer(10987417, 21974834)
 */
function customerCustomizer(customerId, customizerAttributeId) {
    return `customers/${customerId}/customerCustomizers/${customizerAttributeId}`;
}
exports.customerCustomizer = customerCustomizer;
/**
 * @param {string | number} customerId
 * @param {string | number} extensionType
 * @returns `CustomerExtensionSettingResourceName`
 * @example const customerExtensionSetting: ResourceNames.CustomerExtensionSettingResourceName = ResourceNames.customerExtensionSetting(10987417, 21974834)
 */
function customerExtensionSetting(customerId, extensionType) {
    return `customers/${customerId}/customerExtensionSettings/${extensionType}`;
}
exports.customerExtensionSetting = customerExtensionSetting;
/**
 * @param {string | number} customerId
 * @param {string | number} feedId
 * @returns `CustomerFeedResourceName`
 * @example const customerFeed: ResourceNames.CustomerFeedResourceName = ResourceNames.customerFeed(10987417, 21974834)
 */
function customerFeed(customerId, feedId) {
    return `customers/${customerId}/customerFeeds/${feedId}`;
}
exports.customerFeed = customerFeed;
/**
 * @param {string | number} customerId
 * @param {string | number} labelId
 * @returns `CustomerLabelResourceName`
 * @example const customerLabel: ResourceNames.CustomerLabelResourceName = ResourceNames.customerLabel(10987417, 21974834)
 */
function customerLabel(customerId, labelId) {
    return `customers/${customerId}/customerLabels/${labelId}`;
}
exports.customerLabel = customerLabel;
/**
 * @param {string | number} customerId
 * @param {string | number} managerCustomerId
 * @param {string | number} managerLinkId
 * @returns `CustomerManagerLinkResourceName`
 * @example const customerManagerLink: ResourceNames.CustomerManagerLinkResourceName = ResourceNames.customerManagerLink(10987417, 21974834, 43949668)
 */
function customerManagerLink(customerId, managerCustomerId, managerLinkId) {
    return `customers/${customerId}/customerManagerLinks/${managerCustomerId}~${managerLinkId}`;
}
exports.customerManagerLink = customerManagerLink;
/**
 * @param {string | number} customerId
 * @param {string | number} criterionId
 * @returns `CustomerNegativeCriterionResourceName`
 * @example const customerNegativeCriterion: ResourceNames.CustomerNegativeCriterionResourceName = ResourceNames.customerNegativeCriterion(10987417, 21974834)
 */
function customerNegativeCriterion(customerId, criterionId) {
    return `customers/${customerId}/customerNegativeCriteria/${criterionId}`;
}
exports.customerNegativeCriterion = customerNegativeCriterion;
/**
 * @param {string | number} customerId
 * @param {string | number} clusterId
 * @returns `CustomerSearchTermInsightResourceName`
 * @example const customerSearchTermInsight: ResourceNames.CustomerSearchTermInsightResourceName = ResourceNames.customerSearchTermInsight(10987417, 21974834)
 */
function customerSearchTermInsight(customerId, clusterId) {
    return `customers/${customerId}/customerSearchTermInsights/${clusterId}`;
}
exports.customerSearchTermInsight = customerSearchTermInsight;
/**
 * @param {string | number} customerId
 * @param {string | number} accountLinkId
 * @returns `CustomerSkAdNetworkConversionValueSchemaResourceName`
 * @example const customerSkAdNetworkConversionValueSchema: ResourceNames.CustomerSkAdNetworkConversionValueSchemaResourceName = ResourceNames.customerSkAdNetworkConversionValueSchema(10987417, 21974834)
 */
function customerSkAdNetworkConversionValueSchema(customerId, accountLinkId) {
    return `customers/${customerId}/customerSkAdNetworkConversionValueSchemas/${accountLinkId}`;
}
exports.customerSkAdNetworkConversionValueSchema = customerSkAdNetworkConversionValueSchema;
/**
 * @param {string | number} customerId
 * @param {string | number} userId
 * @returns `CustomerUserAccessResourceName`
 * @example const customerUserAccess: ResourceNames.CustomerUserAccessResourceName = ResourceNames.customerUserAccess(10987417, 21974834)
 */
function customerUserAccess(customerId, userId) {
    return `customers/${customerId}/customerUserAccesses/${userId}`;
}
exports.customerUserAccess = customerUserAccess;
/**
 * @param {string | number} customerId
 * @param {string | number} invitationId
 * @returns `CustomerUserAccessInvitationResourceName`
 * @example const customerUserAccessInvitation: ResourceNames.CustomerUserAccessInvitationResourceName = ResourceNames.customerUserAccessInvitation(10987417, 21974834)
 */
function customerUserAccessInvitation(customerId, invitationId) {
    return `customers/${customerId}/customerUserAccessInvitations/${invitationId}`;
}
exports.customerUserAccessInvitation = customerUserAccessInvitation;
/**
 * @param {string | number} customerId
 * @param {string | number} customizerAttributeId
 * @returns `CustomizerAttributeResourceName`
 * @example const customizerAttribute: ResourceNames.CustomizerAttributeResourceName = ResourceNames.customizerAttribute(10987417, 21974834)
 */
function customizerAttribute(customerId, customizerAttributeId) {
    return `customers/${customerId}/customizerAttributes/${customizerAttributeId}`;
}
exports.customizerAttribute = customizerAttribute;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} base64Placement
 * @returns `DetailPlacementViewResourceName`
 * @example const detailPlacementView: ResourceNames.DetailPlacementViewResourceName = ResourceNames.detailPlacementView(10987417, 21974834, 43949668)
 */
function detailPlacementView(customerId, adGroupId, base64Placement) {
    return `customers/${customerId}/detailPlacementViews/${adGroupId}~${base64Placement}`;
}
exports.detailPlacementView = detailPlacementView;
/**
 * @param {string | number} customerId
 * @param {string | number} detailedDemographicId
 * @returns `DetailedDemographicResourceName`
 * @example const detailedDemographic: ResourceNames.DetailedDemographicResourceName = ResourceNames.detailedDemographic(10987417, 21974834)
 */
function detailedDemographic(customerId, detailedDemographicId) {
    return `customers/${customerId}/detailedDemographics/${detailedDemographicId}`;
}
exports.detailedDemographic = detailedDemographic;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} criterionId
 * @returns `DisplayKeywordViewResourceName`
 * @example const displayKeywordView: ResourceNames.DisplayKeywordViewResourceName = ResourceNames.displayKeywordView(10987417, 21974834, 43949668)
 */
function displayKeywordView(customerId, adGroupId, criterionId) {
    return `customers/${customerId}/displayKeywordViews/${adGroupId}~${criterionId}`;
}
exports.displayKeywordView = displayKeywordView;
/**
 * @param {string | number} customerId
 * @param {string | number} placeholderChainId
 * @param {string | number} distanceBucket
 * @returns `DistanceViewResourceName`
 * @example const distanceView: ResourceNames.DistanceViewResourceName = ResourceNames.distanceView(10987417, 21974834, 43949668)
 */
function distanceView(customerId, placeholderChainId, distanceBucket) {
    return `customers/${customerId}/distanceViews/${placeholderChainId}~${distanceBucket}`;
}
exports.distanceView = distanceView;
/**
 * @param {string | number} customerId
 * @param {string | number} campaignId
 * @param {string | number} base64Category
 * @param {string | number} languageCode
 * @returns `DomainCategoryResourceName`
 * @example const domainCategory: ResourceNames.DomainCategoryResourceName = ResourceNames.domainCategory(10987417, 21974834, 43949668, 87899336)
 */
function domainCategory(customerId, campaignId, base64Category, languageCode) {
    return `customers/${customerId}/domainCategories/${campaignId}~${base64Category}~${languageCode}`;
}
exports.domainCategory = domainCategory;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} searchTermFingerprint
 * @param {string | number} headlineFingerprint
 * @param {string | number} landingPageFingerprint
 * @param {string | number} pageUrlFingerprint
 * @returns `DynamicSearchAdsSearchTermViewResourceName`
 * @example const dynamicSearchAdsSearchTermView: ResourceNames.DynamicSearchAdsSearchTermViewResourceName = ResourceNames.dynamicSearchAdsSearchTermView(10987417, 21974834, 43949668, 87899336, 175798672, 351597344)
 */
function dynamicSearchAdsSearchTermView(customerId, adGroupId, searchTermFingerprint, headlineFingerprint, landingPageFingerprint, pageUrlFingerprint) {
    return `customers/${customerId}/dynamicSearchAdsSearchTermViews/${adGroupId}~${searchTermFingerprint}~${headlineFingerprint}~${landingPageFingerprint}~${pageUrlFingerprint}`;
}
exports.dynamicSearchAdsSearchTermView = dynamicSearchAdsSearchTermView;
/**
 * @param {string | number} customerId
 * @param {string | number} expandedFinalUrlFingerprint
 * @returns `ExpandedLandingPageViewResourceName`
 * @example const expandedLandingPageView: ResourceNames.ExpandedLandingPageViewResourceName = ResourceNames.expandedLandingPageView(10987417, 21974834)
 */
function expandedLandingPageView(customerId, expandedFinalUrlFingerprint) {
    return `customers/${customerId}/expandedLandingPageViews/${expandedFinalUrlFingerprint}`;
}
exports.expandedLandingPageView = expandedLandingPageView;
/**
 * @param {string | number} customerId
 * @param {string | number} trialId
 * @returns `ExperimentResourceName`
 * @example const experiment: ResourceNames.ExperimentResourceName = ResourceNames.experiment(10987417, 21974834)
 */
function experiment(customerId, trialId) {
    return `customers/${customerId}/experiments/${trialId}`;
}
exports.experiment = experiment;
/**
 * @param {string | number} customerId
 * @param {string | number} trialId
 * @param {string | number} trialArmId
 * @returns `ExperimentArmResourceName`
 * @example const experimentArm: ResourceNames.ExperimentArmResourceName = ResourceNames.experimentArm(10987417, 21974834, 43949668)
 */
function experimentArm(customerId, trialId, trialArmId) {
    return `customers/${customerId}/experimentArms/${trialId}~${trialArmId}`;
}
exports.experimentArm = experimentArm;
/**
 * @param {string | number} customerId
 * @param {string | number} feedItemId
 * @returns `ExtensionFeedItemResourceName`
 * @example const extensionFeedItem: ResourceNames.ExtensionFeedItemResourceName = ResourceNames.extensionFeedItem(10987417, 21974834)
 */
function extensionFeedItem(customerId, feedItemId) {
    return `customers/${customerId}/extensionFeedItems/${feedItemId}`;
}
exports.extensionFeedItem = extensionFeedItem;
/**
 * @param {string | number} customerId
 * @param {string | number} feedId
 * @returns `FeedResourceName`
 * @example const feed: ResourceNames.FeedResourceName = ResourceNames.feed(10987417, 21974834)
 */
function feed(customerId, feedId) {
    return `customers/${customerId}/feeds/${feedId}`;
}
exports.feed = feed;
/**
 * @param {string | number} customerId
 * @param {string | number} feedId
 * @param {string | number} feedItemId
 * @returns `FeedItemResourceName`
 * @example const feedItem: ResourceNames.FeedItemResourceName = ResourceNames.feedItem(10987417, 21974834, 43949668)
 */
function feedItem(customerId, feedId, feedItemId) {
    return `customers/${customerId}/feedItems/${feedId}~${feedItemId}`;
}
exports.feedItem = feedItem;
/**
 * @param {string | number} customerId
 * @param {string | number} feedId
 * @param {string | number} feedItemSetId
 * @returns `FeedItemSetResourceName`
 * @example const feedItemSet: ResourceNames.FeedItemSetResourceName = ResourceNames.feedItemSet(10987417, 21974834, 43949668)
 */
function feedItemSet(customerId, feedId, feedItemSetId) {
    return `customers/${customerId}/feedItemSets/${feedId}~${feedItemSetId}`;
}
exports.feedItemSet = feedItemSet;
/**
 * @param {string | number} customerId
 * @param {string | number} feedId
 * @param {string | number} feedItemSetId
 * @param {string | number} feedItemId
 * @returns `FeedItemSetLinkResourceName`
 * @example const feedItemSetLink: ResourceNames.FeedItemSetLinkResourceName = ResourceNames.feedItemSetLink(10987417, 21974834, 43949668, 87899336)
 */
function feedItemSetLink(customerId, feedId, feedItemSetId, feedItemId) {
    return `customers/${customerId}/feedItemSetLinks/${feedId}~${feedItemSetId}~${feedItemId}`;
}
exports.feedItemSetLink = feedItemSetLink;
/**
 * @param {string | number} customerId
 * @param {string | number} feedId
 * @param {string | number} feedItemId
 * @param {string | number} feedItemTargetType
 * @param {string | number} feedItemTargetId
 * @returns `FeedItemTargetResourceName`
 * @example const feedItemTarget: ResourceNames.FeedItemTargetResourceName = ResourceNames.feedItemTarget(10987417, 21974834, 43949668, 87899336, 175798672)
 */
function feedItemTarget(customerId, feedId, feedItemId, feedItemTargetType, feedItemTargetId) {
    return `customers/${customerId}/feedItemTargets/${feedId}~${feedItemId}~${feedItemTargetType}~${feedItemTargetId}`;
}
exports.feedItemTarget = feedItemTarget;
/**
 * @param {string | number} customerId
 * @param {string | number} feedId
 * @param {string | number} feedMappingId
 * @returns `FeedMappingResourceName`
 * @example const feedMapping: ResourceNames.FeedMappingResourceName = ResourceNames.feedMapping(10987417, 21974834, 43949668)
 */
function feedMapping(customerId, feedId, feedMappingId) {
    return `customers/${customerId}/feedMappings/${feedId}~${feedMappingId}`;
}
exports.feedMapping = feedMapping;
/**
 * @param {string | number} customerId
 * @param {string | number} placeholderType
 * @returns `FeedPlaceholderViewResourceName`
 * @example const feedPlaceholderView: ResourceNames.FeedPlaceholderViewResourceName = ResourceNames.feedPlaceholderView(10987417, 21974834)
 */
function feedPlaceholderView(customerId, placeholderType) {
    return `customers/${customerId}/feedPlaceholderViews/${placeholderType}`;
}
exports.feedPlaceholderView = feedPlaceholderView;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} criterionId
 * @returns `GenderViewResourceName`
 * @example const genderView: ResourceNames.GenderViewResourceName = ResourceNames.genderView(10987417, 21974834, 43949668)
 */
function genderView(customerId, adGroupId, criterionId) {
    return `customers/${customerId}/genderViews/${adGroupId}~${criterionId}`;
}
exports.genderView = genderView;
/**
 * @param {string | number} criterionId
 * @returns `GeoTargetConstantResourceName`
 * @example const geoTargetConstant: ResourceNames.GeoTargetConstantResourceName = ResourceNames.geoTargetConstant(10987417)
 */
function geoTargetConstant(criterionId) {
    return `geoTargetConstants/${criterionId}`;
}
exports.geoTargetConstant = geoTargetConstant;
/**
 * @param {string | number} customerId
 * @param {string | number} countryCriterionId
 * @param {string | number} locationType
 * @returns `GeographicViewResourceName`
 * @example const geographicView: ResourceNames.GeographicViewResourceName = ResourceNames.geographicView(10987417, 21974834, 43949668)
 */
function geographicView(customerId, countryCriterionId, locationType) {
    return `customers/${customerId}/geographicViews/${countryCriterionId}~${locationType}`;
}
exports.geographicView = geographicView;
/**
 * @param {string | number} googleAdsField
 * @returns `GoogleAdsFieldResourceName`
 * @example const googleAdsField: ResourceNames.GoogleAdsFieldResourceName = ResourceNames.googleAdsField(10987417)
 */
function googleAdsField(googleAdsField) {
    return `googleAdsFields/${googleAdsField}`;
}
exports.googleAdsField = googleAdsField;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} base64Placement
 * @returns `GroupPlacementViewResourceName`
 * @example const groupPlacementView: ResourceNames.GroupPlacementViewResourceName = ResourceNames.groupPlacementView(10987417, 21974834, 43949668)
 */
function groupPlacementView(customerId, adGroupId, base64Placement) {
    return `customers/${customerId}/groupPlacementViews/${adGroupId}~${base64Placement}`;
}
exports.groupPlacementView = groupPlacementView;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} criterionId
 * @returns `HotelGroupViewResourceName`
 * @example const hotelGroupView: ResourceNames.HotelGroupViewResourceName = ResourceNames.hotelGroupView(10987417, 21974834, 43949668)
 */
function hotelGroupView(customerId, adGroupId, criterionId) {
    return `customers/${customerId}/hotelGroupViews/${adGroupId}~${criterionId}`;
}
exports.hotelGroupView = hotelGroupView;
/**
 * @param {string | number} customerId
 * @returns `HotelPerformanceViewResourceName`
 * @example const hotelPerformanceView: ResourceNames.HotelPerformanceViewResourceName = ResourceNames.hotelPerformanceView(10987417)
 */
function hotelPerformanceView(customerId) {
    return `customers/${customerId}/hotelPerformanceView`;
}
exports.hotelPerformanceView = hotelPerformanceView;
/**
 * @param {string | number} customerId
 * @param {string | number} commissionId
 * @returns `HotelReconciliationResourceName`
 * @example const hotelReconciliation: ResourceNames.HotelReconciliationResourceName = ResourceNames.hotelReconciliation(10987417, 21974834)
 */
function hotelReconciliation(customerId, commissionId) {
    return `customers/${customerId}/hotelReconciliations/${commissionId}`;
}
exports.hotelReconciliation = hotelReconciliation;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} criterionId
 * @returns `IncomeRangeViewResourceName`
 * @example const incomeRangeView: ResourceNames.IncomeRangeViewResourceName = ResourceNames.incomeRangeView(10987417, 21974834, 43949668)
 */
function incomeRangeView(customerId, adGroupId, criterionId) {
    return `customers/${customerId}/incomeRangeViews/${adGroupId}~${criterionId}`;
}
exports.incomeRangeView = incomeRangeView;
/**
 * @param {string | number} customerId
 * @param {string | number} invoiceId
 * @returns `InvoiceResourceName`
 * @example const invoice: ResourceNames.InvoiceResourceName = ResourceNames.invoice(10987417, 21974834)
 */
function invoice(customerId, invoiceId) {
    return `customers/${customerId}/invoices/${invoiceId}`;
}
exports.invoice = invoice;
/**
 * @param {string | number} customerId
 * @param {string | number} keywordPlanId
 * @returns `KeywordPlanResourceName`
 * @example const keywordPlan: ResourceNames.KeywordPlanResourceName = ResourceNames.keywordPlan(10987417, 21974834)
 */
function keywordPlan(customerId, keywordPlanId) {
    return `customers/${customerId}/keywordPlans/${keywordPlanId}`;
}
exports.keywordPlan = keywordPlan;
/**
 * @param {string | number} customerId
 * @param {string | number} keywordPlanAdGroupId
 * @returns `KeywordPlanAdGroupResourceName`
 * @example const keywordPlanAdGroup: ResourceNames.KeywordPlanAdGroupResourceName = ResourceNames.keywordPlanAdGroup(10987417, 21974834)
 */
function keywordPlanAdGroup(customerId, keywordPlanAdGroupId) {
    return `customers/${customerId}/keywordPlanAdGroups/${keywordPlanAdGroupId}`;
}
exports.keywordPlanAdGroup = keywordPlanAdGroup;
/**
 * @param {string | number} customerId
 * @param {string | number} keywordPlanAdGroupKeywordId
 * @returns `KeywordPlanAdGroupKeywordResourceName`
 * @example const keywordPlanAdGroupKeyword: ResourceNames.KeywordPlanAdGroupKeywordResourceName = ResourceNames.keywordPlanAdGroupKeyword(10987417, 21974834)
 */
function keywordPlanAdGroupKeyword(customerId, keywordPlanAdGroupKeywordId) {
    return `customers/${customerId}/keywordPlanAdGroupKeywords/${keywordPlanAdGroupKeywordId}`;
}
exports.keywordPlanAdGroupKeyword = keywordPlanAdGroupKeyword;
/**
 * @param {string | number} customerId
 * @param {string | number} keywordPlanCampaignId
 * @returns `KeywordPlanCampaignResourceName`
 * @example const keywordPlanCampaign: ResourceNames.KeywordPlanCampaignResourceName = ResourceNames.keywordPlanCampaign(10987417, 21974834)
 */
function keywordPlanCampaign(customerId, keywordPlanCampaignId) {
    return `customers/${customerId}/keywordPlanCampaigns/${keywordPlanCampaignId}`;
}
exports.keywordPlanCampaign = keywordPlanCampaign;
/**
 * @param {string | number} customerId
 * @param {string | number} keywordPlanCampaignKeywordId
 * @returns `KeywordPlanCampaignKeywordResourceName`
 * @example const keywordPlanCampaignKeyword: ResourceNames.KeywordPlanCampaignKeywordResourceName = ResourceNames.keywordPlanCampaignKeyword(10987417, 21974834)
 */
function keywordPlanCampaignKeyword(customerId, keywordPlanCampaignKeywordId) {
    return `customers/${customerId}/keywordPlanCampaignKeywords/${keywordPlanCampaignKeywordId}`;
}
exports.keywordPlanCampaignKeyword = keywordPlanCampaignKeyword;
/**
 * @param {string | number} expressCategoryId
 * @param {string | number} expressSubCategoryId
 * @returns `KeywordThemeConstantResourceName`
 * @example const keywordThemeConstant: ResourceNames.KeywordThemeConstantResourceName = ResourceNames.keywordThemeConstant(10987417, 21974834)
 */
function keywordThemeConstant(expressCategoryId, expressSubCategoryId) {
    return `keywordThemeConstants/${expressCategoryId}~${expressSubCategoryId}`;
}
exports.keywordThemeConstant = keywordThemeConstant;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} criterionId
 * @returns `KeywordViewResourceName`
 * @example const keywordView: ResourceNames.KeywordViewResourceName = ResourceNames.keywordView(10987417, 21974834, 43949668)
 */
function keywordView(customerId, adGroupId, criterionId) {
    return `customers/${customerId}/keywordViews/${adGroupId}~${criterionId}`;
}
exports.keywordView = keywordView;
/**
 * @param {string | number} customerId
 * @param {string | number} labelId
 * @returns `LabelResourceName`
 * @example const label: ResourceNames.LabelResourceName = ResourceNames.label(10987417, 21974834)
 */
function label(customerId, labelId) {
    return `customers/${customerId}/labels/${labelId}`;
}
exports.label = label;
/**
 * @param {string | number} customerId
 * @param {string | number} unexpandedFinalUrlFingerprint
 * @returns `LandingPageViewResourceName`
 * @example const landingPageView: ResourceNames.LandingPageViewResourceName = ResourceNames.landingPageView(10987417, 21974834)
 */
function landingPageView(customerId, unexpandedFinalUrlFingerprint) {
    return `customers/${customerId}/landingPageViews/${unexpandedFinalUrlFingerprint}`;
}
exports.landingPageView = landingPageView;
/**
 * @param {string | number} criterionId
 * @returns `LanguageConstantResourceName`
 * @example const languageConstant: ResourceNames.LanguageConstantResourceName = ResourceNames.languageConstant(10987417)
 */
function languageConstant(criterionId) {
    return `languageConstants/${criterionId}`;
}
exports.languageConstant = languageConstant;
/**
 * @param {string | number} customerId
 * @param {string | number} leadFormUserSubmissionId
 * @returns `LeadFormSubmissionDataResourceName`
 * @example const leadFormSubmissionData: ResourceNames.LeadFormSubmissionDataResourceName = ResourceNames.leadFormSubmissionData(10987417, 21974834)
 */
function leadFormSubmissionData(customerId, leadFormUserSubmissionId) {
    return `customers/${customerId}/leadFormSubmissionData/${leadFormUserSubmissionId}`;
}
exports.leadFormSubmissionData = leadFormSubmissionData;
/**
 * @param {string | number} customerId
 * @param {string | number} lifeEventId
 * @returns `LifeEventResourceName`
 * @example const lifeEvent: ResourceNames.LifeEventResourceName = ResourceNames.lifeEvent(10987417, 21974834)
 */
function lifeEvent(customerId, lifeEventId) {
    return `customers/${customerId}/lifeEvents/${lifeEventId}`;
}
exports.lifeEvent = lifeEvent;
/**
 * @param {string | number} customerId
 * @param {string | number} campaignId
 * @param {string | number} criterionId
 * @returns `LocationViewResourceName`
 * @example const locationView: ResourceNames.LocationViewResourceName = ResourceNames.locationView(10987417, 21974834, 43949668)
 */
function locationView(customerId, campaignId, criterionId) {
    return `customers/${customerId}/locationViews/${campaignId}~${criterionId}`;
}
exports.locationView = locationView;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} criterionId
 * @returns `ManagedPlacementViewResourceName`
 * @example const managedPlacementView: ResourceNames.ManagedPlacementViewResourceName = ResourceNames.managedPlacementView(10987417, 21974834, 43949668)
 */
function managedPlacementView(customerId, adGroupId, criterionId) {
    return `customers/${customerId}/managedPlacementViews/${adGroupId}~${criterionId}`;
}
exports.managedPlacementView = managedPlacementView;
/**
 * @param {string | number} customerId
 * @param {string | number} mediaFileId
 * @returns `MediaFileResourceName`
 * @example const mediaFile: ResourceNames.MediaFileResourceName = ResourceNames.mediaFile(10987417, 21974834)
 */
function mediaFile(customerId, mediaFileId) {
    return `customers/${customerId}/mediaFiles/${mediaFileId}`;
}
exports.mediaFile = mediaFile;
/**
 * @param {string | number} customerId
 * @param {string | number} merchantCenterId
 * @returns `MerchantCenterLinkResourceName`
 * @example const merchantCenterLink: ResourceNames.MerchantCenterLinkResourceName = ResourceNames.merchantCenterLink(10987417, 21974834)
 */
function merchantCenterLink(customerId, merchantCenterId) {
    return `customers/${customerId}/merchantCenterLinks/${merchantCenterId}`;
}
exports.merchantCenterLink = merchantCenterLink;
/**
 * @param {string | number} mobileAppCategoryId
 * @returns `MobileAppCategoryConstantResourceName`
 * @example const mobileAppCategoryConstant: ResourceNames.MobileAppCategoryConstantResourceName = ResourceNames.mobileAppCategoryConstant(10987417)
 */
function mobileAppCategoryConstant(mobileAppCategoryId) {
    return `mobileAppCategoryConstants/${mobileAppCategoryId}`;
}
exports.mobileAppCategoryConstant = mobileAppCategoryConstant;
/**
 * @param {string | number} criterionId
 * @returns `MobileDeviceConstantResourceName`
 * @example const mobileDeviceConstant: ResourceNames.MobileDeviceConstantResourceName = ResourceNames.mobileDeviceConstant(10987417)
 */
function mobileDeviceConstant(criterionId) {
    return `mobileDeviceConstants/${criterionId}`;
}
exports.mobileDeviceConstant = mobileDeviceConstant;
/**
 * @param {string | number} customerId
 * @param {string | number} offlineUserDataUpdateId
 * @returns `OfflineUserDataJobResourceName`
 * @example const offlineUserDataJob: ResourceNames.OfflineUserDataJobResourceName = ResourceNames.offlineUserDataJob(10987417, 21974834)
 */
function offlineUserDataJob(customerId, offlineUserDataUpdateId) {
    return `customers/${customerId}/offlineUserDataJobs/${offlineUserDataUpdateId}`;
}
exports.offlineUserDataJob = offlineUserDataJob;
/**
 * @param {string | number} criterionId
 * @returns `OperatingSystemVersionConstantResourceName`
 * @example const operatingSystemVersionConstant: ResourceNames.OperatingSystemVersionConstantResourceName = ResourceNames.operatingSystemVersionConstant(10987417)
 */
function operatingSystemVersionConstant(criterionId) {
    return `operatingSystemVersionConstants/${criterionId}`;
}
exports.operatingSystemVersionConstant = operatingSystemVersionConstant;
/**
 * @param {string | number} customerId
 * @param {string | number} campaignId
 * @param {string | number} adGroupId
 * @param {string | number} base64SearchTerm
 * @returns `PaidOrganicSearchTermViewResourceName`
 * @example const paidOrganicSearchTermView: ResourceNames.PaidOrganicSearchTermViewResourceName = ResourceNames.paidOrganicSearchTermView(10987417, 21974834, 43949668, 87899336)
 */
function paidOrganicSearchTermView(customerId, campaignId, adGroupId, base64SearchTerm) {
    return `customers/${customerId}/paidOrganicSearchTermViews/${campaignId}~${adGroupId}~${base64SearchTerm}`;
}
exports.paidOrganicSearchTermView = paidOrganicSearchTermView;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} criterionId
 * @returns `ParentalStatusViewResourceName`
 * @example const parentalStatusView: ResourceNames.ParentalStatusViewResourceName = ResourceNames.parentalStatusView(10987417, 21974834, 43949668)
 */
function parentalStatusView(customerId, adGroupId, criterionId) {
    return `customers/${customerId}/parentalStatusViews/${adGroupId}~${criterionId}`;
}
exports.parentalStatusView = parentalStatusView;
/**
 * @param {string | number} customerId
 * @param {string | number} paymentsAccountId
 * @returns `PaymentsAccountResourceName`
 * @example const paymentsAccount: ResourceNames.PaymentsAccountResourceName = ResourceNames.paymentsAccount(10987417, 21974834)
 */
function paymentsAccount(customerId, paymentsAccountId) {
    return `customers/${customerId}/paymentsAccounts/${paymentsAccountId}`;
}
exports.paymentsAccount = paymentsAccount;
/**
 * @param {string | number} customerId
 * @param {string | number} placeId
 * @returns `PerStoreViewResourceName`
 * @example const perStoreView: ResourceNames.PerStoreViewResourceName = ResourceNames.perStoreView(10987417, 21974834)
 */
function perStoreView(customerId, placeId) {
    return `customers/${customerId}/perStoreViews/${placeId}`;
}
exports.perStoreView = perStoreView;
/**
 * @param {string | number} countryCode
 * @param {string | number} level
 * @param {string | number} id
 * @returns `ProductBiddingCategoryConstantResourceName`
 * @example const productBiddingCategoryConstant: ResourceNames.ProductBiddingCategoryConstantResourceName = ResourceNames.productBiddingCategoryConstant(10987417, 21974834, 43949668)
 */
function productBiddingCategoryConstant(countryCode, level, id) {
    return `productBiddingCategoryConstants/${countryCode}~${level}~${id}`;
}
exports.productBiddingCategoryConstant = productBiddingCategoryConstant;
/**
 * @param {string | number} customerId
 * @param {string | number} adgroupId
 * @param {string | number} criterionId
 * @returns `ProductGroupViewResourceName`
 * @example const productGroupView: ResourceNames.ProductGroupViewResourceName = ResourceNames.productGroupView(10987417, 21974834, 43949668)
 */
function productGroupView(customerId, adgroupId, criterionId) {
    return `customers/${customerId}/productGroupViews/${adgroupId}~${criterionId}`;
}
exports.productGroupView = productGroupView;
/**
 * @param {string | number} customerId
 * @param {string | number} productLinkId
 * @returns `ProductLinkResourceName`
 * @example const productLink: ResourceNames.ProductLinkResourceName = ResourceNames.productLink(10987417, 21974834)
 */
function productLink(customerId, productLinkId) {
    return `customers/${customerId}/productLinks/${productLinkId}`;
}
exports.productLink = productLink;
/**
 * @param {string | number} qualifyingQuestionId
 * @returns `QualifyingQuestionResourceName`
 * @example const qualifyingQuestion: ResourceNames.QualifyingQuestionResourceName = ResourceNames.qualifyingQuestion(10987417)
 */
function qualifyingQuestion(qualifyingQuestionId) {
    return `qualifyingQuestions/${qualifyingQuestionId}`;
}
exports.qualifyingQuestion = qualifyingQuestion;
/**
 * @param {string | number} customerId
 * @param {string | number} recommendationId
 * @returns `RecommendationResourceName`
 * @example const recommendation: ResourceNames.RecommendationResourceName = ResourceNames.recommendation(10987417, 21974834)
 */
function recommendation(customerId, recommendationId) {
    return `customers/${customerId}/recommendations/${recommendationId}`;
}
exports.recommendation = recommendation;
/**
 * @param {string | number} customerId
 * @param {string | number} remarketingActionId
 * @returns `RemarketingActionResourceName`
 * @example const remarketingAction: ResourceNames.RemarketingActionResourceName = ResourceNames.remarketingAction(10987417, 21974834)
 */
function remarketingAction(customerId, remarketingActionId) {
    return `customers/${customerId}/remarketingActions/${remarketingActionId}`;
}
exports.remarketingAction = remarketingAction;
/**
 * @param {string | number} customerId
 * @param {string | number} campaignId
 * @param {string | number} adGroupId
 * @param {string | number} query
 * @returns `SearchTermViewResourceName`
 * @example const searchTermView: ResourceNames.SearchTermViewResourceName = ResourceNames.searchTermView(10987417, 21974834, 43949668, 87899336)
 */
function searchTermView(customerId, campaignId, adGroupId, query) {
    return `customers/${customerId}/searchTermViews/${campaignId}~${adGroupId}~${query}`;
}
exports.searchTermView = searchTermView;
/**
 * @param {string | number} customerId
 * @param {string | number} sharedSetId
 * @param {string | number} criterionId
 * @returns `SharedCriterionResourceName`
 * @example const sharedCriterion: ResourceNames.SharedCriterionResourceName = ResourceNames.sharedCriterion(10987417, 21974834, 43949668)
 */
function sharedCriterion(customerId, sharedSetId, criterionId) {
    return `customers/${customerId}/sharedCriteria/${sharedSetId}~${criterionId}`;
}
exports.sharedCriterion = sharedCriterion;
/**
 * @param {string | number} customerId
 * @param {string | number} sharedSetId
 * @returns `SharedSetResourceName`
 * @example const sharedSet: ResourceNames.SharedSetResourceName = ResourceNames.sharedSet(10987417, 21974834)
 */
function sharedSet(customerId, sharedSetId) {
    return `customers/${customerId}/sharedSets/${sharedSetId}`;
}
exports.sharedSet = sharedSet;
/**
 * @param {string | number} customerId
 * @returns `ShoppingPerformanceViewResourceName`
 * @example const shoppingPerformanceView: ResourceNames.ShoppingPerformanceViewResourceName = ResourceNames.shoppingPerformanceView(10987417)
 */
function shoppingPerformanceView(customerId) {
    return `customers/${customerId}/shoppingPerformanceView`;
}
exports.shoppingPerformanceView = shoppingPerformanceView;
/**
 * @param {string | number} customerId
 * @param {string | number} campaignId
 * @param {string | number} query
 * @returns `SmartCampaignSearchTermViewResourceName`
 * @example const smartCampaignSearchTermView: ResourceNames.SmartCampaignSearchTermViewResourceName = ResourceNames.smartCampaignSearchTermView(10987417, 21974834, 43949668)
 */
function smartCampaignSearchTermView(customerId, campaignId, query) {
    return `customers/${customerId}/smartCampaignSearchTermViews/${campaignId}~${query}`;
}
exports.smartCampaignSearchTermView = smartCampaignSearchTermView;
/**
 * @param {string | number} customerId
 * @param {string | number} campaignId
 * @returns `SmartCampaignSettingResourceName`
 * @example const smartCampaignSetting: ResourceNames.SmartCampaignSettingResourceName = ResourceNames.smartCampaignSetting(10987417, 21974834)
 */
function smartCampaignSetting(customerId, campaignId) {
    return `customers/${customerId}/smartCampaignSettings/${campaignId}`;
}
exports.smartCampaignSetting = smartCampaignSetting;
/**
 * @param {string | number} customerId
 * @param {string | number} customerLinkId
 * @returns `ThirdPartyAppAnalyticsLinkResourceName`
 * @example const thirdPartyAppAnalyticsLink: ResourceNames.ThirdPartyAppAnalyticsLinkResourceName = ResourceNames.thirdPartyAppAnalyticsLink(10987417, 21974834)
 */
function thirdPartyAppAnalyticsLink(customerId, customerLinkId) {
    return `customers/${customerId}/thirdPartyAppAnalyticsLinks/${customerLinkId}`;
}
exports.thirdPartyAppAnalyticsLink = thirdPartyAppAnalyticsLink;
/**
 * @param {string | number} topicId
 * @returns `TopicConstantResourceName`
 * @example const topicConstant: ResourceNames.TopicConstantResourceName = ResourceNames.topicConstant(10987417)
 */
function topicConstant(topicId) {
    return `topicConstants/${topicId}`;
}
exports.topicConstant = topicConstant;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} criterionId
 * @returns `TopicViewResourceName`
 * @example const topicView: ResourceNames.TopicViewResourceName = ResourceNames.topicView(10987417, 21974834, 43949668)
 */
function topicView(customerId, adGroupId, criterionId) {
    return `customers/${customerId}/topicViews/${adGroupId}~${criterionId}`;
}
exports.topicView = topicView;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} criterionId
 * @returns `TravelActivityGroupViewResourceName`
 * @example const travelActivityGroupView: ResourceNames.TravelActivityGroupViewResourceName = ResourceNames.travelActivityGroupView(10987417, 21974834, 43949668)
 */
function travelActivityGroupView(customerId, adGroupId, criterionId) {
    return `customers/${customerId}/travelActivityGroupViews/${adGroupId}~${criterionId}`;
}
exports.travelActivityGroupView = travelActivityGroupView;
/**
 * @param {string | number} customerId
 * @returns `TravelActivityPerformanceViewResourceName`
 * @example const travelActivityPerformanceView: ResourceNames.TravelActivityPerformanceViewResourceName = ResourceNames.travelActivityPerformanceView(10987417)
 */
function travelActivityPerformanceView(customerId) {
    return `customers/${customerId}/travelActivityPerformanceViews`;
}
exports.travelActivityPerformanceView = travelActivityPerformanceView;
/**
 * @param {string | number} customerId
 * @param {string | number} userInterestId
 * @returns `UserInterestResourceName`
 * @example const userInterest: ResourceNames.UserInterestResourceName = ResourceNames.userInterest(10987417, 21974834)
 */
function userInterest(customerId, userInterestId) {
    return `customers/${customerId}/userInterests/${userInterestId}`;
}
exports.userInterest = userInterest;
/**
 * @param {string | number} customerId
 * @param {string | number} userListId
 * @returns `UserListResourceName`
 * @example const userList: ResourceNames.UserListResourceName = ResourceNames.userList(10987417, 21974834)
 */
function userList(customerId, userListId) {
    return `customers/${customerId}/userLists/${userListId}`;
}
exports.userList = userList;
/**
 * @param {string | number} customerId
 * @param {string | number} countryCriterionId
 * @param {string | number} isTargetingLocation
 * @returns `UserLocationViewResourceName`
 * @example const userLocationView: ResourceNames.UserLocationViewResourceName = ResourceNames.userLocationView(10987417, 21974834, 43949668)
 */
function userLocationView(customerId, countryCriterionId, isTargetingLocation) {
    return `customers/${customerId}/userLocationViews/${countryCriterionId}~${isTargetingLocation}`;
}
exports.userLocationView = userLocationView;
/**
 * @param {string | number} customerId
 * @param {string | number} videoId
 * @returns `VideoResourceName`
 * @example const video: ResourceNames.VideoResourceName = ResourceNames.video(10987417, 21974834)
 */
function video(customerId, videoId) {
    return `customers/${customerId}/videos/${videoId}`;
}
exports.video = video;
/**
 * @param {string | number} customerId
 * @param {string | number} adGroupId
 * @param {string | number} criterionId
 * @returns `WebpageViewResourceName`
 * @example const webpageView: ResourceNames.WebpageViewResourceName = ResourceNames.webpageView(10987417, 21974834, 43949668)
 */
function webpageView(customerId, adGroupId, criterionId) {
    return `customers/${customerId}/webpageViews/${adGroupId}~${criterionId}`;
}
exports.webpageView = webpageView;
